src/python/grpcio_tests/tests/unit/_compression_test.py

```diff
@@ -52,75 +113,248 @@ def __init__(self, request_streaming, response_streaming):
+            client_function = _unary_stream_client
+        elif client_streaming and not server_streaming:
+            client_function = _stream_unary_client
+        else:
+            client_function = _stream_stream_client
+
+        server_kwargs = {
+            'compression': grpc.compression.Deflate,
+        } if server_compressed else {}
+        server_handler = _GenericHandler(
+            set_call_compression
+        ) if server_call_compressed else _GenericHandler(None)
+        sent_ratio, received_ratio = _get_compression_ratios(
+            client_function, {}, {}, {}, _GenericHandler(None), channel_kwargs,
+            multicallable_kwargs, server_kwargs, server_handler, _REQUEST)
+
+        if client_side_compressed:
+            self.assertCompressed(sent_ratio)
+        else:
+            self.assertNotCompressed(sent_ratio)
+
+        if server_side_compressed:
+            self.assertCompressed(received_ratio)
+        else:
+            self.assertNotCompressed(received_ratio)
+
+    def testDisableNextCompressionStreaming(self):
+        server_kwargs = {
+            'compression': grpc.compression.Deflate,
+        }
+        _, received_ratio = _get_compression_ratios(
+            _unary_stream_client, {}, {}, {}, _GenericHandler(None), {}, {},
+            server_kwargs, _GenericHandler(disable_next_compression), _REQUEST)
+        self.assertNotCompressed(received_ratio)
+
+
+def _get_compression_str(name, value):
+    if value:
+        return '{}Compressed'.format(name)
+    else:
+        return '{}Uncompressed'.format(name)
+
+
+def _get_compression_test_name(client_streaming, server_streaming,
+                               channel_compressed, multicallable_compressed,
+                               server_compressed, server_call_compressed):
+    client_arity = 'Stream' if client_streaming else 'Unary'
+    server_arity = 'Stream' if server_streaming else 'Unary'
+    arity = '{}{}'.format(client_arity, server_arity)
+    channel_compression_str = _get_compression_str('Channel',
+                                                   channel_compressed)
+    multicallable_compression_str = _get_compression_str(
+        'Multicallable', multicallable_compressed)
+    server_compressed_str = _get_compression_str('Server', server_compressed)
+    server_call_compressed_str = _get_compression_str('ServerCall',
+                                                      server_call_compressed)
+    return 'test{}{}{}{}{}'.format(
+        arity, channel_compression_str, multicallable_compression_str,
+        server_compressed_str, server_call_compressed_str)
+
+
+for client_streaming in (True, False):
+    for server_streaming in (True, False):
+        for channel_compressed in (True, False):
+            for multicallable_compressed in (True, False):
+                for server_compressed in (True, False):
+                    for server_call_compressed in (True, False):
```

Comment:
optional: I like the testing metrics here. I think if you can make them a generator function, it will be even cooler!